infra/getting_started/your_first_lambda_function.md

```diff
@@ -0,0 +1,185 @@
+        runtime          = "go1.x"
+    }
+    ```
+
+    `filename` points to where our `.zip` file is.
+
+    `function_name` is the name given to the function. We can invoke our Lambda function by calling this name.
+
+    `role` is the IAM role attached to this function. We will be creating this in the next step.
+
+    `handler` is the entrypoint into the function. In this particular case, it's the `main.go` file in `.zip`.
+
+    `runtime` specifies what language this function will be running in.
+
+    `source_code_hash` is used to trigger updates when your source code changes.
+
+5. What's left to do finish `main.tf` is creating that IAM role.
+    ```
+    resource "aws_iam_role" "iam_for_lambda_tf" {
+        name = "iam_for_lambda_tf"
+
+        assume_role_policy = <<EOF
+        {
+        "Version": "2012-10-17",
+        "Statement": [
+            {
+            "Action": "sts:AssumeRole",
+            "Principal": {
+                "Service": "lambda.amazonaws.com"
+            },
+            "Effect": "Allow",
+            "Sid": ""
+            }
+        ]
+        }
+    EOF
+    }
+    ```
+
+    This particular function doesn't need access to other AWS services so we're giving it a role with a no access policy.
+
+6. Save the file.
+7. Back on the command line, it's time to initialize the setup for your Terraform plan.
+   ```
+   terraform init
+   ```
+8. If that's looking good, let's check out what our plan looks like.
+    ```
+    terraform plan
+    ```
+9. Provided there's nothing unsurprising in the plan, it's time to deploy your function!
+    ```
+    terraform apply
+    ```
+
+## Invoking your Lambda function
+1. To first check if you deployed successfully, let's print out a list of the existing Lambda functions.
+    ```
+    aws lambda list-functions
+    ```
+2. Once you confirm your function is within that list, invoke your function!
+   ```
+   aws lambda invoke --function-name=test_lambda output.txt
+   ```
+
+3. Congrats, you deployed your first Lambda function with Terraform!!!
+
+## Bonus: Set up a recurring schedule for your Lambda Function
+Let's say you want your shiny new Lambda function to run every day. Maybe every hour! To do this, you will need to use another AWS service called [CloudWatch](https://aws.amazon.com/cloudwatch/). Let's open our `main.tf` up again and add a few things:
+
+1. Add a CloudWatch rule that specifies how often the event will trigger. [This article](https://docs.aws.amazon.com/AmazonCloudWatch/latest/events/ScheduledEvents.html) has more info on how to use the `schedule_expression` value.
+    ```
+    resource "aws_cloudwatch_event_rule" "every_hour" {
+      name = "every-hour"
+      description = "Fires every hour"
+      schedule_expression = "rate(1 hour)"
+    }
+    ```
+2. Specify a target, our Lambda function.
+    ```
+    resource "aws_cloudwatch_event_target" "run_test_lambda_every_hour" {
+      rule = '${aws_cloudwatch_event_rule.every_hour.name}'
```

Comment:
Oh wait, this should be in double quotes.

Reply:
Good catch.